addons/html_editor/static/src/core/dom_plugin.js

```diff
@@ -236,14 +236,21 @@ export class DomPlugin extends Plugin {
                 // Split blocks at the edges if inserting new blocks (preventing
                 // <p><p>text</p></p> or <li><li>text</li></li> scenarios).
                 while (
-                    currentNode.parentElement !== this.editable &&
+                    !this.isEditionBoundary(currentNode.parentElement) &&
                     (!allowsParagraphRelatedElements(currentNode.parentElement) ||
                         currentNode.parentElement.nodeName === "LI")
                 ) {
                     if (isUnbreakable(currentNode.parentElement)) {
-                        makeContentsInline(container);
-                        nodeToInsert = container.childNodes[0];
-                        break;
+                        // If we have to insert a table, we cannot afford to unwrap it
+                        // we need to search for a more suitable spot to put the table in
+                        if (nodeToInsert.nodeName !== "TABLE") {
```

Comment:
It's really up to personal preference, but if you're about to handle the TABLE case right below your comment, I would do 
```
if ( nodeToInsert.nodeName === "TABLE") {...}
else {...}
```